test/correctness/reorder_rvars.cpp

```diff
@@ -24,7 +24,9 @@ int main(int argc, char **argv) {
 
         // Reorder g
         g.reorder(y, x);
-        g.update(0).reorder(r1.x, y, x, r1.y);
+        // It is legal to reorder r1.x and r1.y
+        // because stage g.update(0) is associative.
+        g.update(0).reorder(r1.y, y, x, r1.x);
```

Comment:
Might be worth adding a negative test in test/errors for the is-not-associative case

Reply:
There is an "error_bad_rvar_order" test that still works with this PR. I think it gives an okay coverage. Let me know if you want me to add another test.

Reply:
Nope, that's good enough